### PR TITLE
Use ResizeObserver for map resizes

### LINKED ISSUE #8222: Use ResizeObserver to deprecate L.Map's invalidateSize()
Pretty straightforward feature - attach a `ResizeObserver` to the map's container, and call `invalidateSize()` whenever the container is resized. Detach the `ResizeObserver` when the map is destroyed (on `L.Map`'s `remove()`).

This won't work on any version of IE, nor Edge < 79 (that includes all the Edge versions previous to the switch to chrome's blink engine).

### PR DESCRIPTION
**Pull request by IvanSanchez**
Fixes #8222.

Implementation is pretty straightforward - replaces the old `resize` event handler on `Window`. It also feels nice to get rid of a 9-year-old hack.

As noted in #8222, this won't work in MSIE nor Edge < 79 - but we are not targeting those platforms anymore, so it should be fine.

### CODE REVIEW

**Comment by Falke-Design** [comment]
Maybe a good time to implement a test?

**Review by jonkoops — APPROVED** [review]
Code changes look good, just wondering if this already covered in the test suite.

**Comment by IvanSanchez** [comment]
I'll try and squeeze a couple of test in the next couple of weeks.

**Comment by Falke-Design** [comment]
@IvanSanchez @jonkoops I added some tests, after approval we can merge this PR

**Review by jonkoops — APPROVED** [review]
LGTM. Great change!

**Comment by IvanSanchez** [comment]
@Falke-design I still want to add a test chat changes the CSS properties of the map's container and checks for a `resize` event.

File: spec/suites/map/MapSpec.js [hunk 1]
```diff
@@ -1317,6 +1317,68 @@ describe("Map", () => {
...
 		});
+
+		it("auto invalidateSize after container resize", (done) => {
+			map.setView([0, 0], 0);
+			const spy = sinon.spy();
+			map.on("resize", spy);
+
+			expect(spy.called).to.not.be.ok();
+
+			map.getContainer().style.width = '200px';
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:1328]: @IvanSanchez always fired over css size change?

Comment (IvanSanchez) [reply spec/suites/map/MapSpec.js:1328]: @Falke-Design I'm half asleep and totally skimmed over that really important bit. Damn.

**Comment by MartijnHarmenzon** [comment]
Is there a code example on how to implement this? Can this help to avoid the follwoinf?
```
setTimeout(() => {
  map01.invalidateSize();
}, 100);
```

**Comment by MartijnHarmenzon** [comment]
> Is there a code example on how to implement this? Can this help to avoid the following?
> 
> ```
> setTimeout(() => {
>   map01.invalidateSize();
> }, 100);
> ```

To answer my own question:
```
// Resize observer
const map01Elem = document.getElementById('map01');

const resizeObserver = new ResizeObserver((entries) => {
  map01.invalidateSize(true);
});

resizeObserver.observe(map01Elem);
```

**Pull request merged**